src/SpectralUnmixing.jl

```diff
@@ -45,42 +45,70 @@ export initiate_output_datasets, set_band_names, write_results
 # Unmixing and simulation functions
 export unmix_line, unmix_pixel, simulate_pixel, unmix_and_write_line
 
+"""
+    wl_index(wavelengths::Vector{Float64}, target::Float64)
+
+Return the index of the wavelength in a vector that is closest to the specified target
+wavelength.
+"""
 function wl_index(wavelengths::Vector{Float64}, target)
     argmin(abs.(wavelengths .- target))
 end
 
+"""
+    scale_data(refl::Matrix{Float64}, wavelengths::Vector{Float64}, criteria::String)
+
+Scale the reflectance data based on specified criteria.
+
+# Arguments
+- `refl::Matrix{Float64}`: Reflectance data either size (n,) or (m,n), where n is the
+number of bands and m the number of pixels.
+- `wavelengths::Vector{Float64}`: A vector of wavelengths of size (n,).
+- `criteria::String`: Normalization options:
+  - `"none"`: No scaling will be applied, and the original reflectance data will be
+  returned.
+  - `"brightness"`: The data will be normalized based on brightness, excluding specified
+  bad wavelength regions.
+  - A specific wavelength (as a string): The data will be normalized using the reflectance
+  value at this wavelength.
+
+# Returns
+- A matrix of scaled reflectance data with same dimensions as `refl`.
+"""
 function scale_data(refl::Matrix{Float64}, wavelengths::Vector{Float64}, criteria::String)
 
     if criteria == "none"
         return refl
     elseif criteria == "brightness"
-        bad_regions_wl = [[1300,1500],[1800,2000]]
+        bad_regions_wl = [[1300, 1500], [1800, 2000]]
```

Comment:
Do we want to hardcode these regions in here?

Reply:
Let's convert this to an input to this function as an optional argument, with these populated as defaults.

Reply:
Done, and updated docstring.